pkg/deploy/api/validation/validation.go

```diff
@@ -255,29 +257,18 @@ func validateImageChangeParams(params *deployapi.DeploymentTriggerImageChangePar
 	errs := fielderrors.ValidationErrorList{}
 
 	if len(params.From.Name) != 0 {
-		if len(params.From.Kind) == 0 {
-			params.From.Kind = "ImageStream"
-		}
 		kinds := sets.NewString("ImageRepository", "ImageStream", "ImageStreamTag")
 		if !kinds.Has(params.From.Kind) {
```

Comment:
Also note that we don't allow DockerImages here and in conversion we were converting [DockerImages to DockerImages](https://github.com/openshift/origin/blob/0d4de0b1801e8c58a28c8bcd9d12b5d0c46de1cb/pkg/deploy/api/v1/conversion.go#L143) so that is broken already? I want to test it locally.